DQMOffline/Trigger/plugins/TopMonitor.cc

```diff
@@ -575,17 +577,35 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
       edm::LogWarning("TopMonitor") << "Electron handle not valid \n";
       return;
   }
+  
+  //ATHER                                                                                                                                                                                                                       
+  edm::Handle<edm::ValueMap<bool> > eleIDHandle;
+  iEvent.getByToken(elecIDToken_,  eleIDHandle);
+  if (!eleIDHandle.isValid()){
+    edm::LogWarning("TopMonitor") << "Electron ID handle not valid \n";
+    return;
+  }
+
+
   std::vector<reco::GsfElectron> electrons;
   if (nelectrons_>0){
-      if ( eleHandle->size() < nelectrons_ ) return;
-      for ( auto const & e : *eleHandle ) {
-          if (eleSelection_(e)) electrons.push_back(e);
-          //Suvankar
-          if ( usePVcuts_ &&
-               (std::fabs(e.gsfTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(e.gsfTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
-      }
-      if ( electrons.size() < nelectrons_ ) return;
+    if ( eleHandle->size() < nelectrons_ ) return;
+    size_t index=0;                          //ATHER
+    for ( auto const & e : *eleHandle ) {
```

Comment:
hi @mohsinwaseem - given the changes proposed, this loop should just become a for ( unsigned int i=0; i<eleHandle->size(); i++)... 

otherwise, no need to add your name to each line of code. git will keep track of authorship for you.

Reply:
Hi,
The index here is not for the loop, but to keep track of the object in order to read the correct elecID. 

I added my name in order to make it easy for changes in future. I can remove it in any next PR concerning this code.